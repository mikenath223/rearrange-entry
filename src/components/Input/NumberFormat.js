import NumberFormat from 'react-number-format';
import PropTypes from 'prop-types';

const CustomNumberFormat = ({
  inputRef, onChange, name, ...other
}) => (
  <NumberFormat
    getInputRef={inputRef}
    onValueChange={values => {
      onChange({
        target: {
          name,
          value: values.value,
        },
      });
    }}
    type="tel"
    mask="_"
    format="(###) ###-####"
    isNumericString
    role="input"
    {...other}
  />
);

CustomNumberFormat.propTypes = {
  inputRef: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default CustomNumberFormat;
