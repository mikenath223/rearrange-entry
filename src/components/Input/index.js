import PropTypes from 'prop-types';
import { TextField } from '@material-ui/core';
import CustomNumberFormat from './NumberFormat';

const Input = ({ entry, handleSetEntry }) => (
  <TextField
    value={entry}
    onChange={e => handleSetEntry(e.target.value)}
    name="phone-number"
    label="Enter phone number"
    variant="outlined"
    InputProps={{
      inputComponent: CustomNumberFormat,
    }}
  />
);

Input.propTypes = {
  entry: PropTypes.string.isRequired,
  handleSetEntry: PropTypes.func.isRequired,
};

export default Input;
