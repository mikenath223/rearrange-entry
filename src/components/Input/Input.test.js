import { render } from '@testing-library/react';
import Input from './index';

describe('Input', () => {
  const props = {
    entry: '',
    handleSetEntry: jest.fn(),
  };

  it('should match snapshot', () => {
    const { container } = render(<Input {...props} />);

    expect(container).toMatchSnapshot();
  });
});
