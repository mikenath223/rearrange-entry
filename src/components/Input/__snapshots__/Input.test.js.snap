// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink="false"
    >
      Enter phone number
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        inputmode="numeric"
        role="input"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-3"
        >
          <span>
            Enter phone number
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
